contrib/perf.cc

```diff
@@ -0,0 +1,205 @@
+#include <sourcemeta/jsontoolkit/json.h>
+#include <sourcemeta/jsontoolkit/jsonl.h>
+#include <sourcemeta/jsontoolkit/jsonschema.h>
+
+#include <sourcemeta/blaze/compiler.h>
+#include <sourcemeta/blaze/evaluator.h>
+
+#include <linux/perf_event.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/syscall.h>
+
+#include <algorithm>  // std::sort, std::max
+#include <cstdlib>    // EXIT_SUCCESS, EXIT_FAILURE
+#include <filesystem> // std::filesystem::path
+#include <iostream>   // std::cerr, std::cout
+#include <map>        // std::map
+#include <utility>    // std::pair
+#include <vector>     // std::vector
+
+static long perf_event_open(struct perf_event_attr *hw_event, pid_t pid,
+                            int cpu, long group_fd, unsigned long flags) {
+  long fd;
+  fd = syscall(SYS_perf_event_open, hw_event, pid, cpu, group_fd, flags);
```

Comment:
I wonder if we should not test that `hw_event` is not empty.

Reply:
It's only called within this one file and guaranteed to be initialized correctly, so I don't think this is a concern.